crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -576,14 +572,20 @@ unsafe fn get_component_and_ticks(
             // SAFE: archetypes only store valid table_rows and the stored component type is T
             Some((
                 components.get_unchecked(table_row),
-                components.get_ticks_unchecked(table_row),
+                components.get_ticks_unchecked(table_row) as *const ComponentTicks,
```

Comment:
Should `get_ticks_unchecked` return `*const ComponentTicks`?

Reply:
There are plenty situations where you actually want `&ComponentTicks`, so not necessarily. Ideally I would like to get rid of this pointer cast as well, but that warrants it's own PR if it happens. I tried, but compiled didn't liked it.

Reply:
Oh, it returns `&ComponentTicks`. I assumed that it was `*mut ComponentTicks`.